pkg/controller/component/dependent.go

```diff
@@ -4,14 +4,78 @@ import (
 	"context"
 	routev1 "github.com/openshift/api/route/v1"
 	"github.com/snowdrop/component-operator/pkg/apis/component/v1alpha2"
+	"github.com/snowdrop/component-operator/pkg/controller"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/api/extensions/v1beta1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+type DependentResource interface {
+	Prototype() runtime.Object
```

Comment:
Why do you `Prototype` as name/word instead of `KubernetesObject` as this is what the `runtime.Object` represents ?

Reply:
Because it's meant to be used a a prototype to create other objects, not as an object to use as-is.

Reply:
Can you then document that within the type definition to make it clear why it is called `Prototype` ?